src/otx/engine/engine.py

```diff
@@ -156,6 +153,8 @@ def train(
         callbacks: list[Callback] | Callback | None = None,
         logger: Logger | Iterable[Logger] | bool | None = None,
         resume: bool = False,
+        run_hpo: bool = False,
+        hpo_config: HpoConfig | None = None,
```

Comment:
```suggestion
        hpo_config: HpoConfig = HpoConfig(),
```

Reply:
<img width="304" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/92974184/c3efedd5-b2c1-4b5a-9649-e89c6e716b93">

Object instantiated as initial argument is reused continuously, which has risk to lead to unexpected result as above.